azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DeploymentApiLiveTest.java

```diff
@@ -0,0 +1,127 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import org.jclouds.azurecompute.arm.domain.Deployment;
+import org.jclouds.azurecompute.arm.domain.Deployment.ProvisioningState;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "DeploymentApiLiveTest", singleThreaded = true)
+public class DeploymentApiLiveTest extends BaseAzureComputeApiLiveTest {
+    private int maxTestDuration = 90;
+    private int pollingInterval = 3; // how frequently to poll for create status
+
+    private String resourceGroup;
+    private String resourceName;
+    private String deploymentName;
+
+    @BeforeClass
+    @Override
+    public void setup() {
+        super.setup();
+        resourceGroup = getResourceGroupName();
+    }
+
+    private String getPutBody(String template, String mode, String parameters) {
+        String body = "{ " +
+                      "\"properties\" : " +
+                      "  { " +
+                      "    \"template\" : " + template + ", " +
+                      "    \"mode\" : \"" + mode + "\", " +
+                      "    \"parameters\" : " + parameters + " " +
+                      "  } " +
+                      "}";
+        return body;
+    }
+
+    @Test(groups = "live")
+    public void testCreate() {
+        Long now = System.currentTimeMillis();
+        resourceName = "jcres" + now;
+        deploymentName = "jcdep" + now;
+        String template = "{\"$schema\":\"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#\",\"contentVersion\":\"1.0.0.0\",\"parameters\":{\"newStorageAccountName\":{\"type\":\"string\",\"metadata\":{\"description\":\"Name of the Storage Account\"}},\"storageAccountType\":{\"type\":\"string\",\"defaultValue\":\"Standard_LRS\",\"allowedValues\":[\"Standard_LRS\",\"Standard_GRS\",\"Standard_ZRS\"],\"metadata\":{\"description\":\"Storage Account type\"}},\"location\":{\"type\":\"string\",\"allowedValues\":[\"East US\",\"West US\",\"West Europe\",\"East Asia\",\"Southeast Asia\"],\"metadata\":{\"description\":\"Location of storage account\"}}},\"resources\":[{\"type\":\"Microsoft.Storage/storageAccounts\",\"name\":\"[parameters('newStorageAccountName')]\",\"apiVersion\":\"2015-05-01-preview\",\"location\":\"[parameters('location')]\",\"properties\":{\"accountType\":\"[parameters('storageAccountType')]\"}}]}";
+        String parameters = "{\"newStorageAccountName\":{\"value\":\"" + resourceName + "\"},\"storageAccountType\":{\"value\":\"Standard_LRS\"},\"location\":{\"value\":\"West US\"}}";
+        String properties = getPutBody(template, "Incremental", parameters);
```

Comment:
Yes, I understand that. What I mean is that a deployment template is actually a json that goes directly in the request body. Despite the fact that it is an ARM template (conceptually), there is no practical differente with other API requests. You are just sending a request with a json body.

But the user experience, in this case is worse: Users have to build on their own that json. They can use the builders, but they have to manually serialize the template into a json, just because our `DeploymentApi` methods expect the template to be a json String.

What I wanted to say is that the DeploymentApi methods should be changed to receive a "DeploymentTemplate" (or whatever name it is appropriate) object, instead of a raw json, as it does now with the "properties" string. Let jclouds serialize the ARM template, but keep users just using high level objects. They shouldn't have to worry about generating raw json.